libmamba/src/core/package_cache.cpp

```diff
@@ -168,7 +169,38 @@ namespace mamba
             }
             else
             {
-                LOG_WARNING << "Package tarball '" << tarball_path.string() << "' is invalid";
```

Comment:
I think this message should stay as a WARNING as we should see it even if we don't ask for verbose mode, the other added messages would stay indeed as TRACE.

Reply:
My perspective is that warnings must be shown only if this user is responsible for some unusual operations they can understand in context. I don't think this applies here (if the state of the cache is corrupted or invalid, the users are not responsible for it, but network failure generally are for instance).

IMO cache invalidations might happen but users should not be aware of them, and the caching mechanism should be responsible for updating the state of the cache (here removing the tarball) without making the overall system inconsistent.

This was my intend with changing this line from a warning to a trace.

Reply:
IMHO even if you are not responsible for it (tarball invalidity) the warning should help the user understand what's going on. Warning logs are also here to provide a better user experience.

Reply:
Warnings are useful if we give users some context and the ability to resolve the situation (which is currently not the case with this one IMO).

In this case, we could mention to the users that there is a cache in place which is used to avoid re-downloading packages and that one of their tarball is corruption, explaining that they can remove it manually or clear the entire caches (_e.g._ using `mamba clean --all`), but this is not something they should be doing or even think about in the first place, isn't it?

Reply:
To me, the user is either using `mamba` at a very high level and will ignore the warning (as it is not an error) or he is familiar with (the) software (in general) and the context is clear enough for him to understand what's happening.
Cleaning the cache is a general enough concept for people to understand and use, especially with the logs you added.

Reply:
I guess the hint given on the error is useful and sufficient in this case.

What do you think?

Reply:
In my opinion, the original line warning should stay.

Reply:
I agree they don't help to solve the problem, but they give a hint whjy mamba is downloading a package again while it should be in cache for instance. But honestly, I'm fine with both options, I don't have a strong opinion on this.

Reply:
The tarball is deleted then, so the problem is fixed by mamba; that's why I think there's no reason to report the invalidity in the first place.

Reply:
Thinking more of this, with the latest URL change, once 2.0.7 is released, users with huge environment could be swamped with a lot of these warnings. Let's merge as is with a trace instead of a warning, we can rollback to the warning later when most of the users have upgraded.